evergreen_lint/rules/prevent_tasks_with_tag_on_variants.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from typing import List, NamedTuple, Set
+
+from evergreen_lint.model import LintError, Rule
+
+
+class TagConfig(NamedTuple):
+    variant_tag_name: str  # Tag name for the buildvariant
+    prevent_task_tag: str  # Tag name for tasks that should be prevented
+    ignored_tasks: List[str] = []  # List of task names to be ignored by this rule
```

Comment:
Can we make `ignored_tasks` optional?

Reply:
I have updated the PR to address your comments.